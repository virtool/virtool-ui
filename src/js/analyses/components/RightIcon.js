import React from "react";
import { Icon, Loader } from "../../base";

export const AnalysisItemRightIcon = ({ canModify, onRemove, ready }) => {
    if (ready) {
        if (canModify) {
            return <Icon name="trash" color="red" onClick={onRemove} style={{ fontSize: "17px" }} />;
        }

        return null;
    }

    return <Loader size="14px" color="primary" />;
};
