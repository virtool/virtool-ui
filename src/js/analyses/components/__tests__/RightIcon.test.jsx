import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { describe, expect, it, vi } from "vitest";
import { renderWithProviders } from "../../../../tests/setupTests";
import { AnalysisItemRightIcon } from "../RightIcon";

describe("<RightIcon />", () => {
    it("should render remove icon when [ready=true] and [canModify=true]", async () => {
        const onRemove = vi.fn();

        const { rerender } = renderWithProviders(
            <AnalysisItemRightIcon
                canModify={true}
                ready={true}
                onRemove={onRemove}
            />,
        );

        const button = screen.getByRole("button");

        expect(button).toHaveClass("fas fa-trash");

        await userEvent.click(button);

        // Should call onRemove when clicked.
        expect(onRemove).toHaveBeenCalled();

        rerender(
            <AnalysisItemRightIcon
                canModify={false}
                ready={true}
                onRemove={onRemove}
            />,
        );

        // Should not be shown if the user can't modify the analysis.
        expect(screen.queryByRole("button")).not.toBeInTheDocument();

        rerender(
            <AnalysisItemRightIcon
                canModify={true}
                ready={false}
                onRemove={onRemove}
            />,
        );

        expect(screen.queryByRole("button")).not.toBeInTheDocument();
        expect(screen.getByLabelText("loading")).toBeInTheDocument();
    });
});
