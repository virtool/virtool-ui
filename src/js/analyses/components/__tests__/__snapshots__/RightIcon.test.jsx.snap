// Vitest Snapshot v1

exports[`<RightIcon /> > should render loader when [ready=false] 1`] = `
<Loader
  color="primary"
  size="14px"
/>
`;

exports[`<RightIcon /> > should render remove icon when [ready=true] and [canModify=true] 1`] = `
<Icon
  color="red"
  name="trash"
  onClick={[MockFunction spy]}
  style={
    {
      "fontSize": "17px",
    }
  }
/>
`;

exports[`<RightIcon /> > should return null when [ready=true] and [canModify=false] 1`] = `""`;
